### PR TITLE
Dev kmr

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Comment by k-rister** [comment]
This needs a bit more testing before being marked available for review, but I figured I would get the ball rolling on CI testing now.  I had tested it quite extensively but then I refactored the commits a bit to make for a more logical commit history so I need to ensure I didn't break anything in doing so.

File: engine/engine-script-library [hunk 1]
```diff
@@ -385,33 +382,27 @@ function get_data() {
...
-                scp_from_controller "$ssh_id_file" "$engine_config_dir/bench-cmds/$cs_type/$cs_id/runtime" "bench-runtime-cmds"
+                scp_from_controller "$ssh_id_file" "$engine_config_dir/bench-cmds/$cs_type/$cs_id/runtime" "bench-runtime-cmds" abort get-data-end
             fi
         fi
         if [ "$cs_type" == "server" ]; then
-            scp_from_controller "$ssh_id_file" "$engine_config_dir/bench-cmds/$cs_type/$cs_id/stop" "bench-stop-cmds"
-            if [ ! -e "bench-stop-cmds" ]; then
-                abort_error "bench cmds file bench-stop-cmds not found. PWD: `/bin/pwd`  LS: `/bin/ls`" get-data-end
-            fi
+            scp_from_controller "$ssh_id_file" "$engine_config_dir/bench-cmds/$cs_type/$cs_id/stop" "bench-stop-cmds" abort get-data-end
```
Comment (HughNhan) [inline engine/engine-script-library:398]: Do you really want to remove the next [ -e "bench-stop-cmds ] code block?

File: rickshaw-run [hunk 2]
```diff
@@ -1829,48 +1829,70 @@ sub prepare_bench_tool_engines() {
...
+                    $$file_spec{'required'} = 1;
+                }
                 my $src_file = $$file_spec{'src'};
                 $src_file =~ s/\%bench-dir\%/$run{'bench-dir'}\//g;
                 $src_file =~ s/\%run-dir\%/$run_dir\//g;
                 $src_file =~ s/\%config-dir\%/$config_dir\//g;
                 my $dest_file = $$file_spec{'dest'};
-                printf FH "src=%s\ndest=%s\n", $src_file, $dest_file;
+                if (-e $src_file) {
+                    debug_log(sprintf("adding  %s for engine type '%s'", $src_file, $cs_type));
```
Comment (HughNhan) [inline rickshaw-run:1841]: Would the extra space between "adding   %s"  ruin its perfect look? :-)

**Review by HughNhan — APPROVED** [review]
LGTM!

File: engine/engine-script-library (hunk 1 above)
Comment (k-rister) [reply engine/engine-script-library:398]: The error handler that is being added to `scp_from_controller` handles this.

**Review by atheurer — APPROVED** [review]

**Pull request merged**